src/main/java/com/minecolonies/colony/buildings/BuildingFarmer.java

```diff
@@ -4,131 +4,260 @@
 
+            for(Field field: tempFields)
+            {
+                ScarecrowTileEntity scarecrow = (ScarecrowTileEntity) world.getTileEntity(field.getID());
+                if(scarecrow == null)
+                {
+                    farmerFields.remove(field);
+                    if (currentField != null && currentField.getID() == field.getID())
+                    {
+                        currentField = null;
+                    }
+                }
+                else
+                {
+                    field.setInventoryField(scarecrow.inventoryField);
+                    if(currentField != null && currentField.getID() == field.getID())
+                    {
+                        currentField.setInventoryField(scarecrow.inventoryField);
+                    }
+                }
+            }
+        }
+    }
 
-        public View(ColonyView c, BlockPos l)
+    /**
+     * Resets the fields to need work again.
+     */
+    public void resetFields()
+    {
+        for(Field field: farmerFields)
         {
-            super(c, l);
+            field.setNeedsWork(true);
+            field.calculateSize(getColony().getWorld(), field.getLocation().down());
         }
+    }
 
-        public com.blockout.views.Window getWindow()
+    @Override
+    public void onDestroyed()
+    {
+        super.onDestroyed();
+        for(Field field: farmerFields)
         {
-            return new WindowHutFarmer(this);
+            field.setTaken(false);
         }
+    }
 
-        @Override
-        public void deserialize(ByteBuf buf)
-        {
-            super.deserialize(buf);
+    @Override
+    public AbstractJob createJob(CitizenData citizen)
+    {
+        return new JobFarmer(citizen);
+    }
 
-            wheat = buf.readInt();
-            potato = buf.readInt();
-            carrot = buf.readInt();
-            melon = buf.readInt();
-            pumpkin = buf.readInt();
+    //we have to update our field from the colony!
+    @Override
+    public void readFromNBT(NBTTagCompound compound)
+    {
+        super.readFromNBT(compound);
+        NBTTagList fieldTagList = compound.getTagList(TAG_FIELDS, Constants.NBT.TAG_COMPOUND);
+        for (int i = 0; i < fieldTagList.tagCount(); ++i)
+        {
+            NBTTagCompound fieldCompound = fieldTagList.getCompoundTagAt(i);
+            Field f = Field.createFromNBT(getColony(), fieldCompound);
+            if (f != null)
+            {
+                farmerFields.add(f);
+            }
         }
     }
 
     @Override
-    public void serializeToView(ByteBuf buf)
+    public void writeToNBT(NBTTagCompound compound)
     {
-        super.serializeToView(buf);
-
-        buf.writeInt(wheat);
-        buf.writeInt(potato);
-        buf.writeInt(carrot);
-        buf.writeInt(melon);
-        buf.writeInt(pumpkin);
+        super.writeToNBT(compound);
+        NBTTagList fieldTagList = new NBTTagList();
+        for (Field f : farmerFields)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'f' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)